src/gui/mainwindow.cpp

```diff
@@ -15,27 +22,81 @@ MainWindow::MainWindow(QWidget* parent)
     ui->centralWidget->layout()->setContentsMargins(0, 0, 0, 0);
 
     auto modelRegistry = std::make_shared<QtNodes::DataModelRegistry>();
+    modelRegistry->registerModel<CanDeviceModel>();
+    modelRegistry->registerModel<CanRawSenderModel>();
+    modelRegistry->registerModel<CanRawViewModel>();
 
-    canDevice = std::make_shared<CanDevice>();
     graphScene = std::make_shared<QtNodes::FlowScene>(modelRegistry);
 
+    connect(graphScene.get(), &QtNodes::FlowScene::nodeCreated, this, &MainWindow::nodeCreatedCallback);
+    connect(graphScene.get(), &QtNodes::FlowScene::nodeDeleted, this, &MainWindow::nodeDeletedCallback);
+    connect(graphScene.get(), &QtNodes::FlowScene::nodeDoubleClicked, this, &MainWindow::nodeDoubleClickedCallback);
+
     setupMdiArea();
     connectToolbarSignals();
     connectMenuSignals();
+}
+
+MainWindow::~MainWindow() { delete graphView; }
+
+void MainWindow::closeEvent(QCloseEvent*) { handleExitAction(); }
+
+void MainWindow::nodeCreatedCallback(QtNodes::Node& node)
+{
+    auto dataModel = node.nodeDataModel();
 
-    canDevice->init("socketcan", "can0");
-    canDevice->start();
+    if (dataModel->name() == "CanRawSenderModel") {
```

Comment:
can we use enum class instead? If so, then we will have model type rather than model name.